src/rosdep2/platforms/debian.py

```diff
@@ -83,6 +84,16 @@ def register_elementary(context):
         context.set_os_override(OS_UBUNTU, context.os_detect.get_codename())
 
 
+def register_mx(context):
+    # MX is an alias for Debian. If elementary is detected and it's
+    # not set as an override, force Debian.
+    (os_name, os_version) = context.get_os_name_and_version()
+    if os_name == OS_MX and not context.os_override:
+        print('rosdep detected OS: [%s] aliasing it to: [%s]' %
+              (OS_MX, OS_DEBIAN), file=sys.stderr)
+        context.set_os_override(OS_DEBIAN, context.os_detect.get_codename())
```

Comment:
From my testing, the codename for MX and Debian are different. You'll need to setup an appropriate aliaing method to remap the codenames through so that it knows which base debian packages are to be expected available.

Reply:
New (squashed) commit on ros-infrastructure/rospkg#167. It does not break compatibility with this one.